esrally/track/track.py

```diff
@@ -214,6 +214,13 @@ def filter(self, source_format=None, target_indices=None):
             filtered.append(d)
         return DocumentCorpus(self.name, filtered)
 
+    def union(self, other):
+        assert self.name == other.name, "Both document corpora must have the same name"
```

Comment:
We've discussed this for other cases before, but would it be more consistent with existing behavior to throw an exception here?

Reply:
I can address this, sure.